src/Entities/Player.h

```diff
@@ -709,4 +712,14 @@ class cPlayer :
 	/** Pins the player to a_Location until Unfreeze() is called.
 	If ManuallyFrozen is false, the player will unfreeze when the chunk is loaded. */
 	void FreezeInternal(const Vector3d & a_Location, bool a_ManuallyFrozen);
+
+	/** Returns how high the liquid is in percent. Used by IsInsideOfBlock */
+	float GetLiquidHeightPercent(NIBBLETYPE a_Meta);
+
+	/** Checks if the player is inside of the block (use only for liquid!) */
+	bool IsInsideOfBlock(BLOCKTYPE a_Block);
```

Comment:
How about renaming to `IsInsideLiquidBlock()` that checks against all liquid types at once?